### PR TITLE
LSR-291 Allow longs in screenlets

### PR DESCRIPTION
**Pull request by nhpatt**

### CODE REVIEW

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/BaseScreenlet.java [hunk 1]
```diff
@@ -216,6 +217,23 @@ protected void onScreenletDetached() {
 
 	protected abstract void onUserAction(String userActionName, I interactor, Object... args);
 
+	protected long getLongFromString(String value) {
+		return getLongFromStringWithDefaultValue(value, 0);
+	}
+
+	protected long getLongFromStringWithDefaultValue(String value, long defaultValue) {
```
Comment (jmnavarro) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/BaseScreenlet.java]: I don't understand the semantics of this method

File: android/library/core/src/main/java/com/liferay/mobile/screens/context/LiferayServerContext.java [hunk 2]
```diff
@@ -23,10 +23,26 @@
...
+
+		long companyId = getValueFromIntegerOrString(resources, R.string.liferay_company_id, companyIdentifier);
+		long groupId = getValueFromIntegerOrString(resources, R.string.liferay_group_id, groupIdentifier);
+
+		String server = resources.getString(R.string.liferay_server);
+
+		loadFromValues(companyId, groupId, server);
+	}
+
+	private static long getValueFromIntegerOrString(final Resources resources, final int stringId, int integerId) {
```
Comment (jmnavarro) [inline android/library/core/src/main/java/com/liferay/mobile/screens/context/LiferayServerContext.java]: This method is not necessary

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/BaseScreenlet.java (hunk 1 above)
Comment (nhpatt) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/BaseScreenlet.java]: Rename to castToLong

File: android/library/core/src/main/java/com/liferay/mobile/screens/context/LiferayServerContext.java (hunk 2 above)
Comment (nhpatt) [inline android/library/core/src/main/java/com/liferay/mobile/screens/context/LiferayServerContext.java]: Removed

**Comment by nhpatt** [comment]
Made some changes, ready to review

**Comment by jmnavarro** [comment]
:+1 can be merged when safeCastToLong() is reviewed

**Pull request merged**